src/main/java/com/github/davidmoten/rx/jdbc/State.java

```diff
@@ -6,8 +6,8 @@
 import java.util.concurrent.atomic.AtomicBoolean;
 
 class State {
-    volatile Connection con;
-    volatile PreparedStatement ps;
-    volatile ResultSet rs;
+    Connection con;
```

Comment:
Don't think you can remove the volatile on these because unsubscription can happen from another thread and they must be visible.